admiral/pkg/clusters/serviceentry.go

```diff
@@ -242,30 +242,44 @@ func modifyServiceEntryForNewServiceOrPod(event admiral.EventType, env string, s
 }
 
 //Does two things;
-//i)  Picks the GTP that was created most recently from the passed in GTP list (GTPs from all clusters)
+//i)  Picks the GTP that was created most recently from the passed in GTP list based on GTP priority label (GTPs from all clusters)
 //ii) Updates the global GTP cache with the selected GTP in i)
 func updateGlobalGtpCache(cache *AdmiralCache, identity, env string, gtps map[string][]*v1.GlobalTrafficPolicy) {
 	defer util.LogElapsedTime("updateGlobalGtpCache", identity, env, "")()
+
 	gtpsOrdered := make([]*v1.GlobalTrafficPolicy, 0)
+	priorityGtpsOrdered := make([]*v1.GlobalTrafficPolicy, 0)
+
 	for _, gtpsInCluster := range gtps {
 		gtpsOrdered = append(gtpsOrdered, gtpsInCluster...)
 	}
+
+	var mostRecentGtp *v1.GlobalTrafficPolicy
 	if len(gtpsOrdered) == 0 {
 		log.Debugf("No GTPs found for identity=%s in env=%s. Deleting global cache entries if any", identity, env)
 		cache.GlobalTrafficCache.Delete(identity, env)
 		return
-	} else if len(gtpsOrdered) > 1 {
-		log.Debugf("More than one GTP found for identity=%s in env=%s.", identity, env)
-		//sort by creation time with most recent at the beginning
-		sort.Slice(gtpsOrdered, func(i, j int) bool {
-			iTime := gtpsOrdered[i].CreationTimestamp
-			jTime := gtpsOrdered[j].CreationTimestamp
-			log.Debugf("GTP sorting identity=%s env=%s name1=%s creationTime1=%v name2=%s creationTime2=%v", identity, env, gtpsOrdered[i].Name, iTime, gtpsOrdered[j].Name, jTime)
-			return iTime.After(jTime.Time)
-		})
-	}
+	} else {
+		for _, matchedGtp := range gtpsOrdered {
+			if matchedGtp.ObjectMeta.Labels[common.GetAdmiralParams().LabelSet.ResourcePriorityKey] == "true" {
```

Comment:
@vrushalijoshi , just a thought. This would mean that any GTP created by "anyone" with `admiral.io/resource-priority: true` would be treated as a higher priority. 
Do you think it would make sense to have a value that could be passed as a param when the admiral starts and we only look for that value in the label?

Reply:
Added fix for priority to value instead of true/false